airbyte-cdk/python/airbyte_cdk/sources/declarative/checks/connection_checker.py

```diff
@@ -15,7 +15,7 @@ class ConnectionChecker(ABC):
     """
 
     @abstractmethod
-    def check_connection(self, source: Source, logger: logging.Logger, config: Mapping[str, Any]) -> Tuple[bool, Any]:
+    def check_connection(self, source: AbstractSource, logger: logging.Logger, config: Mapping[str, Any]) -> Tuple[bool, Any]:
```

Comment:
By not using `DeclarativeSource`, we avoid the circular dependency. That being said while being more accurate and allowing to re-enable type checking from mypy in `CheckStream`, we probably have a bigger design problem we should solve eventually.